src/Tasks/Microsoft.NET.Build.Tasks/WriteAppConfig.cs

```diff
@@ -0,0 +1,171 @@
+
+        /// <summary>
+        /// Path to an intermediate file where we can write the input app.config plus the generated startup supportedRuntime
+        /// </summary>
+        [Required]
+        public ITaskItem OutputAppConfigFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            XDocument doc = LoadAppConfig(AppConfigFile);
+
+            AddSupportedRuntimeToAppconfigFile(doc, TargetFrameworkIdentifier, TargetFrameworkVersion);
+
+            if (File.Exists(OutputAppConfigFile.ItemSpec))
+            {
+                File.Delete(OutputAppConfigFile.ItemSpec);
+            }
+
+            var fileStream = new FileStream(
+                OutputAppConfigFile.ItemSpec,
+                FileMode.Create,
+                FileAccess.Write,
+                FileShare.Read);
+            using (var stream = new StreamWriter(fileStream))
+            {
+                doc.Save(stream);
+            }
+        }
+
+        public static void AddSupportedRuntimeToAppconfigFile(
+            XDocument doc,
+            string targetFrameworkIdentifier,
+            string targetFrameworkVersion)
+        {
+            XElement startupNode = doc.Root
+                                      .Nodes()
+                                      .OfType<XElement>()
+                                      .FirstOrDefault(e => e.Name.LocalName == "startup");
+
+            string runtimeVersion = string.Empty;
+            if (!HasExistingSupportedRuntime(startupNode))
+            {
+                if (TryGetSupportRuntimeNode(
+                    targetFrameworkIdentifier,
+                    targetFrameworkVersion,
+                    runtimeVersion,
+                    out XElement supportedRuntime))
+                {
+                    if (startupNode == null)
+                    {
+                        startupNode = new XElement("startup");
+                        doc.Root.Add(startupNode);
+                    }
+
+                    startupNode.Add(supportedRuntime);
+                }
+            }
+        }
+
+        private static bool HasExistingSupportedRuntime(XElement startupNode)
+        {
+            return startupNode != null
+                  && startupNode.Nodes().OfType<XElement>().Any(e => e.Name.LocalName == "supportedRuntime");
+        }
+
+        //https://github.com/dotnet/docs/blob/master/docs/framework/configure-apps/file-schema/startup/supportedruntime-element.md
+        private static bool TryGetSupportRuntimeNode(
+            string targetFrameworkIdentifier,
+            string targetFrameworkVersion,
+            string runtimeVersion,
+            out XElement supportedRuntime)
+        {
+            supportedRuntime = null;
+
+            if (targetFrameworkIdentifier == ".NETFramework"
+                && NuGetVersion.TryParse(targetFrameworkVersion.TrimStart('v', 'V'), out NuGetVersion parsedVersion))
```

Comment:
TargetFrameworkVersion should parse as System.Version

Reply:
Fixed, used System.Version.Parse